src/replication.c

```diff
@@ -1259,15 +1506,65 @@ update_replica_entry(spec_t *spec, replica_t *replica, int iofd)
 		    replica->zvol_guid);
 		goto error_out_replica;
 	}
+
+	known_replica = is_known_replica(spec, replica);
+
+	//TODO: There is a case where user scaled up the replcias (created new cvr)
+	// but he did not update the desired replication factor in that time one of the
+	// known replica is not connected.(Handeled but need to cross verify)
+
+	/* Below snippet is to update known replica list if replica is not present
+	 * known replica list by making call to volume mgmt on succcess update inmemory
+	 * data structure.
+	 */
+	if (known_replica && !is_replica_exist(spec, replica->replica_id)) {
+		// update known replcia list
+		rc = update_known_replica_list(spec, replica);
```

Comment:
can this done even before creating thread for data connection, i.e, before taking replica lock?